### PR TITLE
Admin councillor contribution

### LINKED ISSUE #1202: Let admins see just the councillors made in a single contribution and the details of the contributor
Currently on the admin page for councillor contributions you just see a big list of all the councillors contributed.

screen shot 2017-07-28 at 3 41 23 pm

They cant seen which councillors are connected to the latest contribution, or who made the contribution.

The admin reviewing this needs to know:

* which councillors were submitted in which contribution
* when then contribution was made (is it new or old?)
* the contributors details, if they've been submitted

Then they can load them into the spreadsheet and thank the person.

### PR DESCRIPTION
**Pull request by hisayohorie**
This is to fix #1202 

This is the current index page for councillor contribution
screen shot 2017-08-10 at 10 30 20 pm

It is showing what we want (contributor name, authority, when the submission is made)
When you click view, it goes to the show page

screen shot 2017-08-10 at 10 31 42 pm

and contributor column is not displaying the contributor name, and I've been struggling to make it happen 😞

### CODE REVIEW

**Comment by hisayohorie** [comment]
I made changes in the admin interface with suggestion from @equivalentideas 
Here is what I wanted to achieve in this PR

* Contributions page shows:
  * contributions sorted with the newest one first.
  * date and time of each contribution, in a human readable format.
  * the name of the authority
  * the name of the contributor
* the single Contribution page shows:
  * the date and time it was submitting in a human readable format.
  * the list of councillors in a table with their names and emails
  * the name and email of the contributor
  * the authority

Councillor contribution page has raw of contributor name, when it is created at, authority and detail view link
screen shot 2017-08-11 at 12 12 59 pm

When you click `view`, it goes to the councillor contribution#show page.
It has 
1) the detail of contributor 
2)the table of suggested councillors made in this submission of councillor contribution.

screen shot 2017-08-11 at 12 13 11 pm

When you click link to the contributor, it reads to contributor table
screen shot 2017-08-11 at 12 13 28 pm

in the councillor contribution#show page, the title is currently `created_at`, which can be more readable format. or do we want something else in the title?

**Comment by equivalentideas** [comment]
The admin pages are looking really good @hisayohorie 👍 

> I was trying SO hard to find a way to show the contributor's name in suggested councillor table, but it maybe not necessarily since the contributor info is showing above

I think you're right, it's better to have the contributor information separate from the suggested councillors above the table—looks good as is 👍 

> also it is probably more useful to see suggested_councillor_id which is differ in each column where as contributor is all the same.

I'm not sure why the suggested_councillor_id would be needed by the admin, so feel free to remove that. If you think it's necessary, that's cool too, leave it in and I think make a comment here explaining why it's necessary so we have that record :)

As we were just talking about in chat, the tests are failing because there's already a factory for the CouncillorContribution in the `master` branch, so the one that's been added here is raising the duplicate error after Travis merges it.

The trick here is to remove that and the changes to the Gemfile and spec_helper.rb and then merge the latest version of master in here.

File: Gemfile [hunk 1]
```diff
@@ -69,8 +69,7 @@ group :test do
   gem 'capybara'
   gem 'poltergeist'
   gem 'database_cleaner'
-  gem 'factory_girl_rails'
-  gem 'factory_girl'
+  gem 'factory_girl_rails', :require => false
```
Comment (equivalentideas) [inline Gemfile]: I think this was meant to be removed in reverting the changes to the test suit?

File: Gemfile.lock [hunk 2]
```diff
@@ -556,7 +556,6 @@ DEPENDENCIES
   dotenv-rails
   email_spec (~> 1.6)
   everypolitician-popolo!
-  factory_girl
```
Comment (equivalentideas) [inline Gemfile.lock]: another one that was meant to be reverted?

File: app/models/contributor.rb [hunk 3]
```diff
@@ -1,3 +1,4 @@
 class Contributor < ActiveRecord::Base
   has_many :councillor_contributions
+  has_many :suggested_councillors, through: :councillor_contributions
```
Comment (equivalentideas) [inline app/models/contributor.rb]: Is this necessary?

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -17,6 +17,7 @@
 require "email_spec"
 require 'rspec/active_model/mocks'
 require 'capybara/poltergeist'
+require 'factory_girl_rails'
```
Comment (equivalentideas) [inline spec/spec_helper.rb]: Is this required?

**Comment by equivalentideas** [comment]
This is looking good 👍 as part of this I think we should also consider removing the current 'Suggested Councillors' admin page. Is it still useful now we have the new improved version?

**Comment by equivalentideas** [comment]
Great work @hisayohorie ! Merging.

**Comment by equivalentideas** [comment]
This is now deployed 🚢 🍾 

screen shot 2017-08-16 at 3 05 08 pm

**Pull request merged**